rebasebot/bot.py

```diff
@@ -149,11 +149,22 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
                             "--ancestry-path", f"{merge_base}..dest/{dest.branch}")
     logging.info("Picking commits: \n%s", commits)
 
+    go_mod_commits = []
+
     for commit in commits.splitlines():
         # Commit contains the message for logging purposes,
         # trim on the first space to get just the commit sha
         sha, commit_message = commit.split(" - ")
 
+        if update_go_modules:
+            # If we find a commit with such name, we know that it is a go mod update commit
+            # We then replace <carry> flag for <drop> flag
+            # and append such commit to a list of commits that we want to prune
+            if commit_message.find("UPSTREAM: <carry>: Updating and vendoring go modules "
+                                   "after an upstream rebase"):
+                commit_message.replace("<carry>", "<drop>")
```

Comment:
This returns the replaced value, it doesn't set it, so perhaps it's easier to just `continue` here as we know we aren't keeping the commit? WDYT?

Reply:
Yep, seems to be better.